src/eval-frame/actions/actions.js

```diff
@@ -152,31 +152,23 @@ export function evalConsoleInput(languageId) {
 
     // dispatch(temporarilySaveRunningCellID(cell.id))
     dispatch(incrementExecutionNumber())
-    try {
-      output = runCodeWithLanguage(state.loadedLanguages[evalLanguageId], code)
-    } catch (e) {
-      output = e
-      // evalStatus = 'ERROR'
-    }
-    // clear the console input and text cache immediately after eval
-    dispatch(updateConsoleText(''))
-    dispatch({ type: 'CLEAR_CONSOLE_TEXT_CACHE' })
 
-    const updateAfterEvaluation = () => {
+    const updateAfterEvaluation = (output) => {
       // const cellProperties = { rendered: true }
       // if (evalStatus === 'ERROR') {
       //   cellProperties.evalStatus = evalStatus
       // }
       // dispatch(updateCellProperties(cell.id, cellProperties))
+      dispatch(updateConsoleText(''))
+      dispatch({ type: 'CLEAR_CONSOLE_TEXT_CACHE' })
       dispatch(appendToEvalHistory(null, code, output))
       dispatch(updateUserVariables())
     }
 
-    const evaluation = Promise.resolve()
-      .then(updateAfterEvaluation)
+    return runCodeWithLanguage(language, code)
+      .then(output => updateAfterEvaluation(output))
```

Comment:
the anonymous function is actually not necessary, since 

```javascript
...
  .then(updateAfterEvaluation)
```
does the same thing (output is what the `runCodeWithLanguage` function outputs, so next in the `then` pipe is `output`)

Reply:
Good catch.  This came from the more complex version of this in `evaluateCodeCell` and I didn't fully simplify.